### PR TITLE
Bugfix/#2766 UBS step 2 and step 3

### PR DESCRIPTION
**Pull request by paslavskyi9roman**
**This Pull request contains fixes and features for the UBS page, especially step 2 and 3**

1. **When the user re-open the address pop-up, the district will be displayed**

зображення

2. **Now buttons in pop-up are no longer differ from step 2 buttons. The extra height on hover has been fixed as well**

зображення

3. **When entrance or corpus is not selected - it won't be displaying at the personal information**

зображення

4. **If this is the first order, the message appears** 
 
зображення

5. **When a user clicks "cancel", a confirmation window will appear**

- On step 1 message will appear on cancel when the amount is greater than 0
- In step 2 message will appear on cancel when the address is added (suggestion: should we always have a confirmation on cancel ?)
- In step 3 message will always appear on the cancel 
 
зображення

6. **Fixed the high priority bug when step 3 wasn't displayed until the user made some changes in step 2**

### CODE REVIEW

File: src/app/main/component/ubs/components/ubs-personal-information/ubs-add-address-pop-up/ubs-add-address-pop-up.component.html [hunk 1]
```diff
@@ -14,11 +14,10 @@
       formControlName="district"
       class="shadow-none form-control"
       matGoogleMapsAutocomplete
-      [country]="'ua'"
       [types]="['geocode']"
-      (onAutocompleteSelected)="onDistrictSelected($event)"
+      [country]="'ua'"
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-personal-information/ubs-add-address-pop-up/ubs-add-address-pop-up.component.html:18]: move it to constant

Comment (paslavskyi9roman) [reply src/app/main/component/ubs/components/ubs-personal-information/ubs-add-address-pop-up/ubs-add-address-pop-up.component.html]: ✅ Done
commit: 816e650812df473d991fd7eeb3834df39451ab82

**Review by AlekOK — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Review by SofiiaBaziv — APPROVED** [review]

**Review by Edw1x — APPROVED** [review]

File: src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts [hunk 2]
```diff
@@ -72,6 +90,10 @@ export class UBSOrderDetailsComponent implements OnInit, OnDestroy {
     });
   }
 
+  getFormValues(): boolean {
+    return this.showTotal > 0 ? true : false;
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts:94]: lets simplify this: return this.showTotal > 0;

File: src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.ts [hunk 3]
```diff
@@ -156,8 +173,14 @@ export class UBSPersonalInformationComponent implements OnInit, OnDestroy {
       .subscribe(() => this.findAllAddresses());
   }
 
+  getFormValues(): boolean {
+    return true;
+  }
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.ts:178]: what the point of this ?

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts [hunk 4]
```diff
@@ -28,6 +46,10 @@ export class UBSSubmitOrderComponent implements OnInit, OnDestroy {
     this.destroy.unsubscribe();
   }
 
+  getFormValues(): boolean {
+    return true;
+  }
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:51]: what the point of this ?

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts [hunk 5]
```diff
@@ -8,16 +12,30 @@ import { UBSOrderFormService } from '../../services/ubs-order-form.service';
...
+    closeOnNavigation: true,
+    disableClose: true,
+    panelClass: 'popup-dialog-container',
+    data: {
+      popupTitle: 'confirmation.title',
+      popupSubtitle: 'confirmation.subTitle',
+      popupConfirm: 'confirmation.cancel',
+      popupCancel: 'confirmation.dismiss'
+    }
+  };
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:34]: can we create one popupConfig and pass it to child component from parent ?

File: src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.scss [hunk 6]
```diff
@@ -167,3 +167,12 @@ label,
   margin: 3px;
   cursor: pointer;
 }
+
+.address {
+  width: 500px;
```
Comment (JimHolman1199) [inline src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.scss:172]: will this work for mobile ?

File: src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.ts (hunk 3 above)
Comment (paslavskyi9roman) [reply src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.ts:178]: the function "getFormValues" effect on the warning window appearing. It does return a boolean. In this particular case (Step 2), I always need to show the warning window.

File: src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts (hunk 4 above)
Comment (paslavskyi9roman) [reply src/app/main/component/ubs/components/ubs-submit-order/ubs-submit-order.component.ts:51]: the function "getFormValues" effect on the warning window appearing. It does return a boolean. In this particular case (Step 3), I always need to show the warning window.

File: src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.scss (hunk 6 above)
Comment (paslavskyi9roman) [reply src/app/main/component/ubs/components/ubs-personal-information/ubs-personal-information.component.scss:172]: **This screenshot was captured on iPhone X/XS width (375px)**

File: src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts (hunk 2 above)
Comment (paslavskyi9roman) [reply src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts]: this line has been shortened with this commit https://github.com/ita-social-projects/GreenCityClient/pull/896/commits/43890ba93b0ec853b4de3ab164c0da4d997db727

File: src/app/main/component/shared/components/form-base/form-base.component.ts [hunk 7]
```diff
@@ -64,4 +65,23 @@ export class FormBaseComponent implements ComponentCanDeactivate {
...
+      const matDialogRef = this.dialog.open(WarningPopUpComponent, this.popupConfig);
+
+      matDialogRef
+        .afterClosed()
+        .pipe(take(1))
+        .subscribe((confirm) => {
+          if (confirm) {
+            this.areChangesSaved = true;
+            this.router.navigate([this.previousPath]);
+          }
```
Comment (JimHolman1199) [inline src/app/main/component/shared/components/form-base/form-base.component.ts:80]: if you need only truthy value - you can filter it with help of rxjs -> https://stackoverflow.com/questions/53953015/using-rxjs-with-filterboolean-for-queries

**Pull request merged**